src/Classes/ExtBuildListControl.lua

```diff
@@ -0,0 +1,551 @@
+		80, "VERTICAL") {
+		-- y = function()
+		-- 	return (self.scrollH and -8 or 0)
+		-- end,
+		height = function()
+			local _, height = self:GetSize()
+			return height - 2 - (self.scrollH and 16 or 0)
+		end
+	}
+	if not self.scroll then
+		self.controls.scrollBarV.shown = false
+	end
+
+	if self.activeListProvider:GetPageUrl() then
+		self.controls.all = new("ButtonControl", { "BOTTOM", self, "BOTTOM" }, 0, 1, self.width, 20, "See All",
+			function()
+				local url = self.activeListProvider:GetPageUrl()
+				if url then
+					OpenURL(url)
+				end
+			end)
+		self.controls.all.width = function()
+			return self.width()
+		end
+	end
+end
+
+function ExtBuildListControlClass:SetImportCode(importCode)
+	self.importCode = importCode
+end
+
+function ExtBuildListControlClass:IsMouseOver()
+	if not self:IsShown() then
+		return
+	end
+	return self:IsMouseInBounds() or self:GetMouseOverControl()
+end
+
+function ExtBuildListControlClass:OnKeyDown(key, doubleClick)
+	if not self:IsShown() or not self:IsEnabled() then
+		return
+	end
+	local mOverControl = self:GetMouseOverControl()
+	if mOverControl and mOverControl.OnKeyDown then
+		return mOverControl:OnKeyDown(key)
+	end
+	if not self:IsMouseOver() and key:match("BUTTON") then
+		return
+	end
+end
+
+function ExtBuildListControlClass:OnKeyUp(key)
+	if not self:IsShown() or not self:IsEnabled() then
+		return
+	end
+
+	if self.controls.scrollBarV:IsScrollDownKey(key) then
+		self.controls.scrollBarV:Scroll(1)
+	elseif self.controls.scrollBarV:IsScrollUpKey(key) then
+		self.controls.scrollBarV:Scroll(-1)
+	end
+
+	if key == "LEFTBUTTON" then
+		self:CheckButtons()
+	end
+end
+
+function ExtBuildListControlClass:GetHoveredButton()
+	if self.inTransition then
+		return
+	end
+	local cursorX, cursorY = GetCursorPos();
+	cursorY = cursorY + self.controls.scrollBarV.offset
+	for i, importButton in ipairs(self.importButtons) do
+		if (cursorX > importButton.x0) and
+			(cursorX < importButton.x1) and
+			(cursorY > importButton.y0) and
+			(cursorY < importButton.y1) then
+			return {
+				button = importButton,
+				type = 'import'
+			}
+		end
+	end
+
+	for i, previewButton in ipairs(self.previewButtons) do
+		if (cursorX > previewButton.x0) and
+			(cursorX < previewButton.x1) and
+			(cursorY > previewButton.y0) and
+			(cursorY < previewButton.y1) then
+			return {
+				button = previewButton,
+				type = 'preview'
+			}
+		end
+	end
+end
+
+function ExtBuildListControlClass:GetAscendancyImageHandle(ascendancy)
+	local image = nil
+	if ascendancy then
+		image = NewImageHandle()
+		image:Load(s_format('Assets/ascendants/%s.jpeg', (ascendancy:gsub("^%l", string.lower))))
+	end
+
+	return image
+end
+
+function ExtBuildListControlClass:GetClassImageHandle(class)
+	local image = nil
+	if class then
+		image = NewImageHandle()
+		image:Load(s_format('Assets/ascendants/%s.jpeg', (class:gsub("^%l", string.lower))))
+	end
+
+	return image
+end
```

Comment:
This is the same function

Reply:
Yeah in my first class fallback implementation I thought I would add something special for classes but I didn't after all. Fixed into single function now.